clusterclient.go

```diff
@@ -0,0 +1,120 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+)
+
+// ClusterClient provides a high-level API for miscellaneous (i.e. non consume or produce) Kafka cluster operations
+type ClusterClient struct {
+	brokers []string
+	dialer  *Dialer
+}
+
+// NewClusterClient creates and returns a *ClusterClient with a []string of bootstrap
+// brokers for connecting to the cluster and the DefaultDialer
+func NewClusterClient(brokers []string) *ClusterClient {
+	return &ClusterClient{
+		brokers: brokers,
+		dialer:  DefaultDialer,
+	}
+}
+
+// NewClusterClient creates and returns a *ClusterClient with a []string of bootstrap
+// brokers for connecting to the cluster and a user supplied Dialer
+func NewClusterClientWith(brokers []string, d *Dialer) *ClusterClient {
+	return &ClusterClient{
+		brokers: brokers,
```

Comment:
Let's make a copy of the list of brokers so we don't risk sharing memory that may get reused and mess up the internal configuration of the client.

Reply:
Done